lib/vector-buffers/src/topology/channel/poll_notify.rs

```diff
@@ -0,0 +1,81 @@
+use futures::ready;
```

Comment:
This is here because `tokio_util` had no such utility, and we needed a way to drive `async fn` futures via a poll-based mechanism for `LimitedSender<T>`/`LimitedReceiver<T>`.

This will go away once we're able to work on removing `Sink` from the buffer topology code entirely.

Reply:
With the changes in https://github.com/vectordotdev/vector/pull/11348, I _think_ we should be able to drop `Sink` if we can provide something on `BufferSender` like tokio's [`async fn reserve(&self)`](https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Sender.html#method.reserve).

Reply:
Yeah, definitely.  Thinking that will end up being a follow-up PR here, given how big this one has gotten overall.